integration/sandbox_run_rollback_test.go

```diff
@@ -229,6 +233,91 @@ func TestRunPodSandboxWithShimStartAndTeardownCNIFailure(t *testing.T) {
 	t.Run("JustCleanup", testCase(false))
 }
 
+// TestRunPodSandboxWithShimStartAndTeardownCNISlow should keep the sandbox
+// record if failed to rollback CNI API.
+func TestRunPodSandboxWithShimStartAndTeardownCNISlow(t *testing.T) {
+	if runtime.GOOS != "linux" {
+		t.Skip()
+	}
+	if os.Getenv("ENABLE_CRI_SANDBOXES") != "" {
+		t.Skip()
+	}
+
+	t.Log("Init PodSandboxConfig with specific key")
+	labels := map[string]string{
+		t.Name(): "true",
+	}
+	sbConfig := PodSandboxConfig(t.Name(), "failpoint", WithPodLabels(labels))
+
+	t.Log("Inject CNI failpoint")
+	conf := &failpointConf{
+		// Delay 10s
+		Add: "1*delay(10000)",
+		// This is to make sure the sandbox container won't be deleted even if the defer functions are invoked in sandbox_run.go
+		// Without this, the test is flaky because ListPodSandbox may return empty list.
+		Del: "1*error(please retry)",
```

Comment:
It does occasionally happens. An example error is as below.

```
failed to setup network for sandbox \"6bf28c849ce9a8fb55833ffa7964ab3cd785e666d129cdfae86e27d97f3b1bc5\": plugin type=\"cni-bridge-fp\" failed (add): netplugin failed with no error message: signal: killed"
```

Talked offline with @fuweid, this may be caused by graceful shutdown during SIGTERM of containerd server stop. The contexts are cancels and sometimes the cni.Add is killed.

I will remove this "Del" injection and use a SIGKILL to restart containerd.